python/housinginsights/ingestion/LoadData.py

```diff
@@ -286,8 +289,267 @@ def rebuild(self):
+                           Column('acs_median_rent', Numeric),
+                           Column('acs_upper_rent_quartile', Numeric))
+
+        # add to db
+        metadata.create_all(tables=[zone_facts])
+
+        # populate table with calculated fields values
+        self._populate_zone_facts_table()
+
+        return metadata.tables
+
+    def _populate_zone_facts_table(self):
+        """
+        Populates the zone_facts table with the calculated fields data and
+        acs rent data fields from census.
+
+        :return: list of SQLAlchemy query result objects from inserting the
+        calculated fields data into the zone_facts table.
+        """
+        census_fields = [
+            'poverty_rate', 'fraction_black', 'income_per_capita',
+            'labor_participation', 'fraction_foreign',
+            'fraction_single_mothers', 'acs_lower_rent_quartile',
+            'acs_median_rent', 'acs_upper_rent_quartile'
+        ]
+
+        zone_types = ['ward', 'neighborhood_cluster', 'census_tract']
+
+        query_results = list()
+
+        # populate columns accordingly for each zone_specific type
+        for zone_type in zone_types:
+            field_values = dict()
+
+            # get field value for each zone_specific type
+            for field in census_fields:
+                result = self._census_with_weighting(data_id=field,
+                                                     grouping=zone_type)
+                field_values[field] = result['items']
+
+            zone_specifics = self._get_zone_specifics_for_zone_type(zone_type)
+
+            # TODO: add aggregate for each zone_type into table
+            for zone in zone_specifics:
+                # get not None values so we can added to db
+                columns = list()
+                values = list()
+                for field in census_fields:
+                    zone_value = field_values[field][zone]
+
+                    if zone_value is not None:
+                        columns.append(field)
+                        values.append("'" + str(zone_value) + "'")
+
+                # derive column and values strings needed for sql query
+                columns = ', '.join(columns)
+                columns = 'zone, ' + columns
+
+                values = ', '.join(values)
+                values = "'" + zone + "', " + values
+
+                q = "INSERT INTO zone_facts ({cols}) VALUES ({vals})".format(
+                    cols=columns, vals=values)
+
+                with self.engine.connect() as conn:
+                    result = conn.execute(q)
+                    query_results.append(result)
+
+        return query_results
+
+    def _get_zone_specifics_for_zone_type(self, zone_type):
+        """
+        Returns a list of zone_specific values for a given zone_type.
+        """
+        with self.engine.connect() as conn:
+
+            if zone_type == 'ward':
+                table = 'project'
```

Comment:
you're right - that would be better just in case in future a ward isn't represented in the project table.